rts/lra/lra-test/lra-test-basic/src/test/java/io/narayana/lra/arquillian/FailingParticipantCallsIT.java

```diff
@@ -90,7 +90,10 @@ public void testFailingAfterLRA() {
             }
         }
 
-        new NarayanaLRARecovery().waitForRecovery(lra);
+        NarayanaLRARecovery recoveryHelper = new NarayanaLRARecovery();
+
+        recoveryHelper.waitForRecovery(lra);
+        recoveryHelper.waitForRecovery(lra);
```

Comment:
May I ask why this fixes the issue? `waitForRecovery` should keep replaying the end phase until the LRA is finished. Is something not expected returned from the first pass? In that case, we should probably fix the NarayanaLRARecovery.

Reply:
The test fails because the callback from the coordinator to the listener is not sent/received in time (we have coded a hard limit of 1 second on callbacks). What happens is that the coordinator sends the callback and either the listener does not respond within the second or the response is not received within a second.

LRA guarantees that it will keep retrying until the listener acknowledges receipt of the callback. Hence the second recovery pass.

Note that the coordinator is correctly issuing the callback but I am unable to reproduce this failure on my laptop (even when I apply heavy load to the CPU) which lends wait to it being a straight forward CI resourcing problem.

Reply:
If I could duplicate the failure on my laptop then I could debug the issue more precisely. This particular PR is theoretical fix based on code inspection.